### PR TITLE
Randomize new partner rather than run whole process

### PR DESCRIPTION
**Pull request by kanin-kearpimy**
Hi, I noticed this process is possibly slowing down compilation time.

### TLDR;
You can jump to [(3)](https://arc.net/l/quote/dhhqcsuc) for more clarification.

### Background
In the mutation of permutation, CryptOpt finds the range of independent operators. Each range is defined by minimum & maximum (min & max).

range

Then, the Optimizer selects **chosen** and **partner** between [min, max] range for instruction scheduling. It results in three possibilities.

1. There is only one operator in the range (min == max). So, it must pick up another range by re-permutation.
range1

2. The operators in the range are only two operators (min + 1 == max). The optimizer can switch between min & max.
range 2

3. More than 2 choices in operators (max - min > 2). In case **chosen** and **partner** are **same** elements, the current program decides to run the whole permutation process. I think this will slow down the mutation time. 

**Solution:** Instead, we are able to **randomize** another potential **partner** in the **same range**.

range 3

### CODE REVIEW

**Comment by dderjoel** [comment]
I like the idea, but before we fix something that *potentially* speeds up something, I wonder if this actually makes things faster. Would you be able to benchmark both versions?

Also, as far as I know it is "TLDR: Too long, didn't read", not TLRD.

Also, I do not understand those intervals in the illustrations. Which instruction is 'chosen' in those pictures?
